### PR TITLE
Update free-courses-hi.md

### PR DESCRIPTION
**Pull request by ShivaySabharwal**
Added Data Structures, and made some structural changes according to the guidelines.

## Hacktoberfest notes:

- due to volume of submissions, we may not be able to review PRs that do not pass tests and do not have informative titles.
- please read our [contributing guidelines](/CONTRIBUTING.md)
- be sure to check the output of Travis-CI for linter errors
- if this is your first open source contribution, make sure it's not your last!

## What does this PR do?
Add Resource(s) | Remove Resource(s) | Add info | Improve Repo

### Checklist:
- [x] Not a duplicate
- [x] Included author(s) if appropriate
- [x] Lists are in alphabetical order
- [x] Needed indications added (PDF, access notes, under construction)

### CODE REVIEW

**Comment by eshellman** [comment]
For youtube, we prefer "playlist" links and the name of the producer/creator.

**Comment by ShivaySabharwal** [comment]
I have made the required changes.

File: free-courses-hi.md [hunk 1]
```diff
@@ -15,16 +17,23 @@
...
 ### Python
 
-* [Python Tutorial For Beginners In Hindi (With Notes)](https://www.youtube.com/watch?v=gfDE2a7MKjA) - Harry
+* [Python Tutorial For Beginners | Hindi (With Notes)](https://www.youtube.com/watch?v=gfDE2a7MKjA) - Harry
 
 
 ### Javascript
 
-* [JavaScript Tutorial In Hindi](https://www.youtube.com/watch?v=hKB-YGF14SY) - Harry
+* [JavaScript Tutorial | Hindi](https://www.youtube.com/watch?v=hKB-YGF14SY) - Harry
```
Comment (eshellman) [inline free-courses-hi.md:39]: no playlist link?

File: free-courses-hi.md [hunk 2]
```diff
@@ -15,16 +17,23 @@
...
+
 ### Java
 
 * [The complete Java tutorial | Hindi](https://www.youtube.com/playlist?list=PLu0W_9lII9agS67Uits0UnJyrYiXhDS6q) - Harry
 
 
 ### Python
 
-* [Python Tutorial For Beginners In Hindi (With Notes)](https://www.youtube.com/watch?v=gfDE2a7MKjA) - Harry
+* [Python Tutorial For Beginners | Hindi (With Notes)](https://www.youtube.com/watch?v=gfDE2a7MKjA) - Harry
```
Comment (eshellman) [inline free-courses-hi.md:34]: no playlist link?

**Comment by ShivaySabharwal** [comment]
Made the changes.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**